### PR TITLE
Remove unused daemon.json keys

### PR DESCRIPTION
**Pull request by ashcrow**
Instead of providing every possible key in the daemon.json configuration
file, this update only provides keys which are actively set/used by the
installer.

This will make it easier to test and explicit to what we use (or don't use) today.

### CODE REVIEW

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ashcrow** [comment]
@ingvagabund added the doc link to README.

File: roles/docker/templates/daemon.json [hunk 1]
```diff
@@ -1,65 +1,20 @@
...
-    "default-gateway-v6": "",
     "default-runtime": "oci",
     "containerd": "/run/containerd.sock",
-    "default-ulimits": {},
     "disable-legacy-registry": false,
-    "dns": [],
-    "dns-opts": [],
-    "dns-search": [],
     "exec-opts": ["native.cgroupdriver=systemd"],
-    "exec-root": "",
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Hmm, wondering why it was set to "" if the doc says "/var/run/docker".

File: roles/docker/templates/daemon.json [hunk 2]
```diff
@@ -1,65 +1,20 @@
...
-    "fixed-cidr": "",
-    "fixed-cidr-v6": "",
-    "graph": "",
-    "group": "",
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: The same here, default is "docker"

File: roles/docker/templates/daemon.json [hunk 3]
```diff
@@ -1,65 +1,20 @@
...
-    "default-ulimits": {},
     "disable-legacy-registry": false,
-    "dns": [],
-    "dns-opts": [],
-    "dns-search": [],
     "exec-opts": ["native.cgroupdriver=systemd"],
-    "exec-root": "",
-    "fixed-cidr": "",
-    "fixed-cidr-v6": "",
-    "graph": "",
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: The same here, default is "/var/lib/docker"

File: roles/docker/templates/daemon.json [hunk 4]
```diff
@@ -1,65 +1,20 @@
...
-    "group": "",
-    "hosts": [],
-    "icc": false,
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Default value of icc is `true`.

File: roles/docker/templates/daemon.json [hunk 5]
```diff
@@ -1,65 +1,20 @@
...
     "insecure-registries": {{ l_docker_insecure_registries }},
-    "ip": "0.0.0.0",
-    "iptables": false,
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Default value is `true`

File: roles/docker/templates/daemon.json [hunk 6]
```diff
@@ -1,65 +1,20 @@
...
-    "ipv6": false,
-    "ip-forward": false,
-    "ip-masq": false,
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Default value is `true`

File: roles/docker/templates/daemon.json [hunk 7]
```diff
@@ -1,65 +1,20 @@
...
-    "fixed-cidr-v6": "",
-    "graph": "",
-    "group": "",
-    "hosts": [],
-    "icc": false,
     "insecure-registries": {{ l_docker_insecure_registries }},
-    "ip": "0.0.0.0",
-    "iptables": false,
-    "ipv6": false,
-    "ip-forward": false,
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Default value is `true`

File: roles/docker/templates/daemon.json [hunk 8]
```diff
@@ -1,65 +1,20 @@
...
 {% if docker_log_driver is defined  %}
     "log-driver": "{{ docker_log_driver }}",
 {%- endif %}
-    "log-level": "",
     "log-opts": {{ l_docker_log_options }},
-    "max-concurrent-downloads": 3,
-    "max-concurrent-uploads": 5,
-    "mtu": 0,
-    "oom-score-adjust": -500,
-    "pidfile": "",
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Default value is `/var/run/docker.pid`

File: roles/docker/templates/daemon.json [hunk 9]
```diff
@@ -1,65 +1,20 @@
...
     "runtimes": {
 	"oci": {
 	    "path": "/usr/libexec/docker/docker-runc-current"
 	}
     },
     "selinux-enabled": {{ l_docker_selinux_enabled | lower }},
-    "storage-driver": "",
-    "storage-opts": [],
-    "tls": true,
-    "tlscacert": "",
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Default value is `~/.docker/ca.pem`

File: roles/docker/templates/daemon.json [hunk 10]
```diff
@@ -1,65 +1,20 @@
...
-    "tlscert": "",
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Default value is `~/.docker/cert.pem`

File: roles/docker/templates/daemon.json [hunk 11]
```diff
@@ -1,65 +1,20 @@
...
-    "tlskey": "",
```
Comment (ingvagabund) [inline roles/docker/templates/daemon.json]: Default value is `~/.docker/key.pem`

**Comment by ashcrow** [comment]
@ingvagabund if a default was changed in docker it would also change in the traditional package install as well.

**Comment by ingvagabund** [comment]
> @ingvagabund if a default was changed in docker it would also change in the traditional package install as well.

That is true. Acked.

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ashcrow** [comment]
[merge]

**Pull request merged**